common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -467,27 +583,16 @@ private void verifySmartcardPin(@NonNull final char[] pin, @NonNull final YubiKi
             piv.verifyPin(pin);
             //If pin is successfully verified, we will get the certificate and perform the rest of the logic for authentication.
             final X509Certificate cert = piv.getCertificate(certDetails.getSlot());
-            //TODO: do a confirmation that the cert retrieved above is the same cert that was picked earlier by comparing with YubiKitCertDetails
-            //TODO: Complete authentication using cert and YubiKit sdk.
-            //TODO: Delete test code below between START and END comments (should be replaced with actual authentication)
-            //NOTE: below is for testing. This would get replaced by the logic for actual authentication.
-            //TODO:START of code for testing
-            Logger.infoPII(methodTag, "Using cert object here as a placeholder for when it's needed for authentication: " + cert.getSubjectDN());
-            mActivity.runOnUiThread(new Runnable() {
-                @Override
-                public void run() {
-                    Toast.makeText(mActivity.getApplicationContext(), "Success!", Toast.LENGTH_LONG).show();
-                    //Reset currentDialog to null
-                    synchronized (sSmartcardDialogLock) {
-                        mCurrentDialog = null;
-                    }
-                }
-            });
-            request.cancel();
-            synchronized (sSmartcardDialogLock) {
-                mCurrentDialog.dismiss();
+            //Do a confirmation that the cert retrieved above is the same cert that was picked earlier by comparing with YubiKitCertDetails.
+            if (!cert.getSubjectDN().getName().equals(certDetails.getSubjectText()) || !cert.getIssuerDN().getName().equals(certDetails.getIssuerText())) {
```

Comment:
was going to say that you should use getEncoded() to compare certs (or compare their thumbprints)... but I don't think you actually need to compare it - if you store X509 in YubiKitCertDetails in the first place.